### DISCUSSION TITLE
[Design/Balance] Should delayed baton knockdown respect antistuns?

### DISCUSSION THREAD

**Discussion by SpringSkipper** [ParadiseSS13/Paradise#20388] [Documents]
I figured I should bring this up here before making any sort of PR. The proposal I had is that the security baton's delayed knockdown would be changed to a status effect, and antagonist antistuns would remove the status to prevent the knockdown. The trait that prevents cycling between batons to bypass their cooldown would remain separate.

## But why??
To my understanding, the delayed knockdown was meant to be more of a softened disable than an instant knockdown would be. But many antistun effects (epi OD, adrenals, rejuv) are coded to _immediately_ remove all stuns and stamloss as well as to (in the case of epi OD and adrenals) regenerate stamina and reduce all subsequent stuns. This creates the situation where a bad man takes a bop, pops an antistun to try to get away, and then falls over five steps down the hall, having been unsuccessful in removing all debilitating effects since, while in its 'delay' stage, the delayed knockdown doesn't count as a disabling status.

I'd like to hear a balance or design team member chime in if this is a fully intended quirk of security batons, but it strikes me as an unintended side effect of that softening of the instant knockdown. The unintuitive result of an ability meant to clear all stuns not clearing all stuns catches a lot of newer players off-guard and complicates security interactions even for experienced players. That second part isn't necessarily a bad thing though, which is why I wanted to start this discussion.

## What would be affected?
This change would affect epi OD, adrenals, rejuv, bath salts (which has a significant drawback and which is by far the most difficult narcotic to make), and the medical-exclusive stimulants injector (which costs 8 TC, is one-use, and has a drawback mechanic that prevents it from easily being spread out over multiple uses).

The delayed knockdown would become a status effect that is specifically removed by these effects upon activation (epi OD, adrenals, rejuv) or upon life process tick (stimulants and bath salts). Disable reduction from things like Pump Up and the lingering effects of epi OD and adrenals would still reduce the eventual knockdown as normal, but would not prevent it from occurring after the delay.

**Comment by Rurikz** [comment]
I spoke a bit on the discord about this and Ill mention here: With the current iteration of stunbatons, the delayed knockdown is what makes them so damn good. You can 1v1 an adrenal user and keep them fully floored the entire duration if you time your baton hits well, denying escape, and to an extent, combat potential. As far as I am aware this is intentional.

I see merit for this, baton being the anti-stun nullifier if you get a hit in melee, fulfilling a role the disabler does not have. As an officer, its very satisfying to deny 8 tc with good play.

However, at the end of the day I don't think its makes for good balance. Aside from being a semi-noob trap (baton, use antistun method, fall over anyway, everyone makes that mistake once), I think it makes for questionable balance. Batons are common, part of base officer kit, and can completely deny escape of any kind if used well. Its led to the issue that I think bmon was trying to get to in his baton PR (just guessing, I havent asked bmon personally), where batons are almost at the top of the totem poll despite commonality. 

Should a baton user be able to 1v1 someone with adrenals with relative ease if they get the first click? Should an adrenal user be perma-floored with no chance to escape vs 2 people with batons? Should 2-3 baton users be the nightmare it is to fight currently, since you'll be floored the entire time no matter what you use, unless its stims or chainsaw?

In the end I think Spring-Skippers mentioned change would be more healthy for the combat dynamic, and make things easier to balance around. I'd @ balance team members for their opinion, but I have no idea who they are. I think this deserves looking over. Please think on it.

**Comment by SpringSkipper** [comment]
Upon further investigation, baton does 60 stamina and has a 2.5 second knockdown delay. This means that, if it were to just be a stamina check, it could result in inconsistent behavior with things like meth where if a life proc occurs within the first 0.5 seconds (recovering 40 stamina) then the next life proc 2 seconds later would recover you to full stamina and cancel the knockdown.

I think code-wise it would be better to separate the baton knockdown callback timer into status effect and cooldown. The cooldown prevents baton cycling while the status effect (unless it gets removed by an antistun) would cause the delayed knockdown.

In this case I'd specifically add code to remove the delayed knockdown to cure-all traitor antistuns (epi OD, stimulants, adrenals, rejuvenate) and also to bath salts (it's the hardest antistun to make, has no speed boost, substantial drawbacks, and is already coded to fully remove all stuns and stamina damage per life tick just like stimulants does).

I'll edit my post accordingly.
